pkg/server/server.go

```diff
@@ -131,12 +130,33 @@ type Server struct {
 	autoUpdateExitCode    int
 }
 
-func New(ctx context.Context, config *lepconfig.Config, endpoint string, cliUID string, packageManager *gpud_manager.Manager, opts ...gpud_config.OpOption) (_ *Server, retErr error) {
-	options := &gpud_config.Op{}
-	if err := options.ApplyOpts(opts); err != nil {
-		return nil, err
-	}
+type createComponentFunc func(context.Context, *lepconfig.Config) components.Component
```

Comment:
We should export the function type, and should return error
```suggestion
type NewComponentFunc func(context.Context, *lepconfig.Config) (components.Component, error)
```